### PR TITLE
PMM-7452: Test storage location credentials

### PR DESCRIPTION
**Pull request by fabio-silva**
What this PR does / why we need it: Allow admins to test the connection to a backup storage location before effectively adding it.

Which issue(s) this PR fixes: https://jira.percona.com/browse/PMM-7452

# Implementation 
- Added "Test" button to modal;
- The button triggers an "onTest" callback, which blocks the "Add"/"Edit" button and sets the "Test" button in a loading state;
- Either a success or an error notification is shown after the API returns

# Screenshots
Screenshot 2021-02-25 at 15 17 56 (2)
Screenshot 2021-02-25 at 15 16 26 2
Screenshot 2021-02-25 at 15 21 45 (2)
Screenshot 2021-02-25 at 15 20 30 (2)

### CODE REVIEW

File: public/app/features/backup/components/StorageLocations/AddStorageLocationModal/AddStorageLocationModal.styles.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { css } from 'emotion';
+
+export const getStyles = () => ({
+  button: css`
+    min-width: 80px;
+    justify-content: center;
+  `,
+  testButton: css`
+    background: linear-gradient(180deg, #04be5b 0%, #04b155 100%);
+    border: 1px solid #058943;
```
Comment (tiagomotasantos) [inline public/app/features/backup/components/StorageLocations/AddStorageLocationModal/AddStorageLocationModal.styles.ts:10]: Also there's a theme variable for this 1px `theme.border.width.sm`

File: public/app/features/backup/components/StorageLocations/AddStorageLocationModal/AddStorageLocationModal.styles.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+    &:hover,
+    &:focus {
+      background: #04be5b;
```
Comment (tiagomotasantos) [inline public/app/features/backup/components/StorageLocations/AddStorageLocationModal/AddStorageLocationModal.styles.ts:14]: Please use a theme variable here, there are some variables for green colors.

**Review by tiagomotasantos — CHANGES_REQUESTED** [review]

File: public/app/features/backup/components/StorageLocations/AddStorageLocationModal/AddStorageLocationModal.styles.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { css } from 'emotion';
+
+export const getStyles = () => ({
+  button: css`
+    min-width: 80px;
+    justify-content: center;
+  `,
+  testButton: css`
+    background: linear-gradient(180deg, #04be5b 0%, #04b155 100%);
```
[the remark this answers is not in the document]
Comment (fabio-silva) [reply public/app/features/backup/components/StorageLocations/AddStorageLocationModal/AddStorageLocationModal.styles.ts]: Sure, good point 👍

File: public/app/features/backup/components/StorageLocations/AddStorageLocationModal/AddStorageLocationModal.styles.ts (hunk 1 above)
Comment (fabio-silva) [reply public/app/features/backup/components/StorageLocations/AddStorageLocationModal/AddStorageLocationModal.styles.ts]: Thanks 👍

File: public/app/features/backup/components/StorageLocations/AddStorageLocationModal/AddStorageLocationModal.styles.ts (hunk 2 above)
Comment (fabio-silva) [reply public/app/features/backup/components/StorageLocations/AddStorageLocationModal/AddStorageLocationModal.styles.ts]: 👍

**Review by tiagomotasantos — APPROVED** [review]

**Review by nicolalamacchia — APPROVED** [review]

**Pull request merged**